src/libostree/ostree-repo-pull.c

```diff
@@ -2133,6 +2195,65 @@ repo_remote_fetch_summary (OstreeRepo    *self,
   return ret;
 }
 
+static void
+delta_superblock_fetch_on_complete (GObject        *object,
+                         GAsyncResult   *result,
+                         gpointer        user_data)
+{
+  OstreeFetcher *fetcher = (OstreeFetcher *)object;
+  FetchDeltaSuperBlockData *fetch_data = user_data;
+  OtPullData *pull_data = fetch_data->pull_data;
+  GError *local_error = NULL;
+  GError **error = &local_error;
+  GInputStream* input = _ostree_fetcher_stream_uri_finish (fetcher, result, error);
+  GBytes* delta_superblock_data = NULL;
+
+  if (input)
+    {
+      delta_superblock_data = g_input_stream_read_bytes ( input, OSTREE_MAX_METADATA_SIZE, pull_data->cancellable, error);
```

Comment:
Actually in the error case here, we may then get into an overwriting-error situation `if (pull_data->require_static_deltas)` etc.

Say that the webserver returns 500 and we're requiring static deltas.

(Hm, we should probably have `ostree trivial-httpd` support a mode where it returns `500`s too).